packages/aws-cdk/lib/api/aws-auth/aws-sdk-inifile.ts

```diff
@@ -54,39 +58,78 @@ export class PatchedSharedIniFileCredentials extends AWS.SharedIniFileCredential
     var externalId = roleProfile.external_id;
     var mfaSerial = roleProfile.mfa_serial;
     var sourceProfileName = roleProfile.source_profile;
+    var sourceCredentials = roleProfile.credential_source;
 
-    if (!sourceProfileName) {
+    if (sourceProfileName && sourceCredentials) {
       throw (AWS as any).util.error(
-        new Error('source_profile is not set using profile ' + this.profile),
+        new Error('source_profile and credential_source are both configured in profile ' + this.profile + '. please choose one or the other.'),
         { code: 'SharedIniFileCredentialsProviderFailure' },
       );
     }
 
-    var sourceProfileExistanceTest = creds[sourceProfileName];
-
-    if (typeof sourceProfileExistanceTest !== 'object') {
+    if (!sourceProfileName && !sourceCredentials) {
       throw (AWS as any).util.error(
-        new Error('source_profile ' + sourceProfileName + ' using profile '
-          + this.profile + ' does not exist'),
+        new Error('neither source_profile nor credential_source configured in profile ' + this.profile + '. please configure one or the other.'),
         { code: 'SharedIniFileCredentialsProviderFailure' },
       );
     }
 
-    var sourceCredentials = new AWS.SharedIniFileCredentials(
-      (AWS as any).util.merge(this.options || {}, {
-        profile: sourceProfileName,
-        preferStaticCredentials: true,
-      }),
-    );
-
     // --------- THIS IS NEW ----------------------
```

Comment:
This comment is not accurate anymore :). Might as well get rid of it.

Reply:
👍